redundant-bmc/src/sibling_impl.cpp

```diff
@@ -0,0 +1,293 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+#include "sibling_impl.hpp"
+
+#include <phosphor-logging/lg2.hpp>
+#include <xyz/openbmc_project/ObjectMapper/client.hpp>
+
+namespace rbmc
+{
+
+bool SiblingImpl::isBMCPresent()
+{
+    // TODO: Actually check this.
+    // May also need to check if it has Vcs PGOOD.
+    return true;
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> SiblingImpl::init()
+{
+    if (initialized)
+    {
+        lg2::info("Sibling::init called more than once");
+        co_return;
+    }
+
+    ctx.spawn(watchInterfaceAdded());
+    ctx.spawn(watchInterfaceRemoved());
+    ctx.spawn(watchPropertyChanged());
+
+    serviceName = co_await getServiceName();
```

Comment:
A few comments throughout this function would be appreciated

Reply:
done